compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaSymbol.java

```diff
@@ -65,7 +67,16 @@ protected BallerinaSymbol(String name, SymbolKind symbolKind, BSymbol symbol, Co
 
     @Override
     public Optional<String> getName() {
-        return Optional.ofNullable(this.name);
+        // In the langlib context, reserved keywords can be used as regular identifiers. Therefore, they will not be
+        // escaped.
+        if (getModule().isPresent()) {
+            ModuleID moduleID = getModule().get().id();
+            if (moduleID != null && !moduleID.moduleName().isEmpty() && moduleID.moduleName().startsWith("lang.") &&
+                    !moduleID.orgName().isEmpty() && moduleID.orgName().startsWith("ballerina")) {
+                return Optional.ofNullable(this.name);
+            }
+        }
+        return Optional.ofNullable(escapeReservedKeyword(this.name));
```

Comment:
The 2 conditions `moduleID != null && !moduleID.moduleName().isEmpty()` are unnecessary IINM. If there's a module symbol, there should always be a module ID. And since we are checking `startsWith()`, checking for an empty string is irrelevant right?

Reply:
Another small point, shall we cache the escaped name? So that we don't have to compute it each time `getName()` is called. A symbol is immutable. So it won't change after creation. Refer to some of the other methods for examples of where this has been done.

Reply:
@pubudu91 thanks for the suggestion. 

Sure, that'd be a great idea to cache it.

Reply:
@pubudu91 used a caching mechanism. Also implemented it in the `BallerinaRecordFieldSymbol` and `BallerinaObjectFieldSymbol` classes as well.